test/unit/reductions/grid_search/test_grid_search_arguments.py

```diff
@@ -231,6 +232,40 @@ def test_grid_size_warning_up_to_5_sensitive_feature_group(self, transformX, tra
         else:
             assert len(log_records) == 0
 
+    @pytest.mark.parametrize("transformX", candidate_X_transforms)
+    @pytest.mark.parametrize("transformY", candidate_Y_transforms)
+    @pytest.mark.parametrize("transformA", candidate_A_transforms)
+    def test_custom_grid(self, transformX, transformY, transformA):
+
+        grid_size = 10
+        grid_limit = 2.0
+        grid_offset = 0.1
+
+        disparity_moment = EqualizedOdds()
+        X, y, A = _quick_data(False)
+
+        disparity_moment.load_data(X, y, sensitive_features=A)
+
+        grid = _GridGenerator(grid_size, grid_limit,
+            disparity_moment.pos_basis, disparity_moment.neg_basis,
+            disparity_moment.neg_basis_present, False, grid_offset).grid
+
+        indices = [7, 3, 4]
+        grid = grid.iloc[:, indices]
```

Comment:
Or is it this bit which matters? An extra comment of two in the test code would be appreciated.

Reply:
Yes, this is the bit that matters, which makes a grid DataFrame whose column names are not ordered integers. I'll add a comment.